src/main/java/edu/ucla/library/prl/harvester/handlers/UpdateJobHandler.java

```diff
@@ -130,6 +136,19 @@ Future<UpdateResponse> updateSolr(final Tuple3<Job, Job, Institution> aData) {
         });
     }
 
+    /**
+     * @param anOldJob A Job representing an original state
+     * @param aNewJob A Job representing a new, updated state
+     * @return True if new job has sets not contained in old job
+     */
+    private boolean hasNewSets(final Job anOldJob, final Job aNewJob) {
```

Comment:
In addition to reworking the implementation:

```suggestion
    private Future<Boolean> hasNewSets(final Job anOldJob, final Job aNewJob) {
```

Reply:
I've figured out how to do `hasNewSets` as a `Future<Boolean>`, but I'm stumped on how to splice it into the sequence of `Future<T>.compose` that starts on line 54 (this branch)...

Reply:
Try something like this:

```java
final Job oldJob = oldJobAndInstitution._1();
final Institution institution = oldJobAndInstitution._2();

return hasNewSets(oldJob, job).compose(itDoes -> {
    final Job jobToSubmit;

    if (itDoes) {
        jobToSubmit = new Job(job.getInstitutionID(), job.getRepositoryBaseURL(), job.getSets() job.getScheduleCronExpression(), null);
    } else {
        jobToSubmit = job;
    }

    final Future<Void> update = ...; // same as what you already have

    return update;
});
```